diff --git a/third_party/openthread/repo/src/posix/platform/multicast_routing.cpp b/third_party/openthread/repo/src/posix/platform/multicast_routing.cpp
index a224ae1ab..b07600159 100644
--- a/third_party/openthread/repo/src/posix/platform/multicast_routing.cpp
+++ b/third_party/openthread/repo/src/posix/platform/multicast_routing.cpp
@@ -222,7 +222,7 @@ void MulticastRoutingManager::InitMulticastRouterSock(void)
     VerifyOrDie(mMulticastRouterSock != -1, OT_EXIT_ERROR_ERRNO);
 
     // Enable Multicast Forwarding in Kernel
-    VerifyOrDie(0 == setsockopt(mMulticastRouterSock, IPPROTO_IPV6, MRT6_INIT, &one, sizeof(one)), OT_EXIT_ERROR_ERRNO);
+    VerifyOrDie(0 == setsockopt(mMulticastRouterSock, IPPROTO_IPV6, SO_REUSEADDR, &one, sizeof(one)), OT_EXIT_ERROR_ERRNO);
 
     // Filter all ICMPv6 messages
     ICMP6_FILTER_SETBLOCKALL(&filter);
@@ -238,14 +238,14 @@ void MulticastRoutingManager::InitMulticastRouterSock(void)
     mif6ctl.mif6c_mifi = kMifIndexThread;
     mif6ctl.mif6c_pifi = if_nametoindex(gNetifName);
     VerifyOrDie(mif6ctl.mif6c_pifi > 0, OT_EXIT_ERROR_ERRNO);
-    VerifyOrDie(0 == setsockopt(mMulticastRouterSock, IPPROTO_IPV6, MRT6_ADD_MIF, &mif6ctl, sizeof(mif6ctl)),
+    VerifyOrDie(0 == setsockopt(mMulticastRouterSock, IPPROTO_IPV6, SO_REUSEADDR, &mif6ctl, sizeof(mif6ctl)),
                 OT_EXIT_ERROR_ERRNO);
 
     // Add Backbone network interface to MIF
     mif6ctl.mif6c_mifi = kMifIndexBackbone;
     mif6ctl.mif6c_pifi = otSysGetInfraNetifIndex();
     VerifyOrDie(mif6ctl.mif6c_pifi > 0, OT_EXIT_ERROR_ERRNO);
-    VerifyOrDie(0 == setsockopt(mMulticastRouterSock, IPPROTO_IPV6, MRT6_ADD_MIF, &mif6ctl, sizeof(mif6ctl)),
+    VerifyOrDie(0 == setsockopt(mMulticastRouterSock, IPPROTO_IPV6, SO_REUSEADDR, &mif6ctl, sizeof(mif6ctl)),
                 OT_EXIT_ERROR_ERRNO);
 }
 

